src/adapter/src/explain_new/mir/text.rs

```diff
@@ -437,6 +449,37 @@ impl<'a> Displayable<'a, MirRelationExpr> {
             writeln!(f)
         }
     }
+
+    pub fn fmt_indexed_filter<'b, C>(
```

Comment:
It's a bit weird that this method has been defind on the type. I think it's a better fit for the virtual syntax pattern adopted by `ExceptAll` in the HIR rendering.

Reply:
`fmt_indexed_filter` is also called from the fast path explain code (`src/adapter/src/coord/peek.rs`), so that's why I made it a method. I didn't find a better place to put it, but I'm open to suggestions.

Orthogonally, I could move the other call of `fmt_indexed_filter` from `fmt_raw_syntax` to `fmt_virtual_syntax`. Should I do that?

Reply:
I played a bit with what I had in mind in 9468872c110bcd1635e17f6879e8ce6edaec18b9 (it's on [my fork of this branch](https://github.com/MaterializeInc/materialize/commit)). 

While I think this is conceptually a good fit for modelling this, there are some practical issues with this approach at the moment. Most notably, the de-constructed fields returned by `AlgIndexedFilter::un_indexed_filter` are not enough construct the backing representation with `AlgIndexedFilter::indexed_filter` (specifically for an `MIR` carrier).

Let's merge this as it is now.

Reply:
Ok, thank you for looking into it!